<a href="https://colab.research.google.com/github/aljubic1/bioinformatika_projekt_alj/blob/main/notebooks/02_feature_engineering.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 02. Feature Engineering

## Cilj

Cilj ovog notebooka je pretvoriti peptidne sekvence iz pripremljenog skupa podataka u numeričke značajke koje modeli strojnog učenja mogu koristiti. Iz svake sekvence izračunat će se duljina, neto naboj, prosječna hidrofobnost i aminokiselinski sastav (AAC).

In [ ]:
# Uvoz biblioteke pandas za rad s tabličnim podacima.
import pandas as pd

# Uvoz sys modula za dodavanje putanje do src direktorija.
import sys

# Dodavanje src direktorija u Python putanju.
sys.path.append("/content")

# Uvoz funkcije za izračun značajki.
from features import extract_features

In [ ]:
# Učitavanje očišćenog skupa podataka.
df = pd.read_csv("/content/amp_clean.csv")

# Prikaz prvih pet redaka.
df.head()

,ID,COMPLEXITY,NAME,N TERMINUS,SEQUENCE,C TERMINUS,SYNTHESIS TYPE,TARGET GROUP,TARGET OBJECT,label
0,8,Monomer,NaN,C16,KVVVKWVVKVVK,AMD,Synthetic,Gram+ Gram- Fungus,Lipid Bilayer,1
1,10,Monomer,NaN,NaN,LFIFFF,AMD,Synthetic,Gram+,Lipid Bilayer,1
2,11,Monomer,"Cathelicidin-1, CATH-1, Fowlicidin-1",NaN,RVKRVWPLVIRTVIAGYNLYRAIKKK,NaN,Ribosomal,Gram+ Gram- Cancer Mammalian Cell,Lipid Bilayer,1
3,12,Monomer,NaN,NaN,RKRIHIGPGRAFYTT,NaN,Synthetic,Gram+ Gram-,Lipid Bilayer,1
4,14,Monomer,Brevinin-2 related peptide B2RP,NaN,GIWDTIKSMGKVFAGKILQNL,AMD,Ribosomal,Gram+ Gram- Virus Fungus Mammalian...,Lipid Bilayer Virus replication,1


In [ ]:
# Izračun bioinformatičkih značajki.
features_df = extract_features(df)

# Prikaz prvih pet redaka.
features_df.head()

,ID,SEQUENCE,length,net_charge,hydrophobicity,label,AAC_A,AAC_C,AAC_D,AAC_E,...,AAC_M,AAC_N,AAC_P,AAC_Q,AAC_R,AAC_S,AAC_T,AAC_V,AAC_W,AAC_Y
0,8,KVVVKWVVKVVK,12,4,1.075000,1,0.000000,0.0,0.000000,0.0,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.583333,0.083333,0.000000
1,10,LFIFFF,6,0,3.250000,1,0.000000,0.0,0.000000,0.0,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
2,11,RVKRVWPLVIRTVIAGYNLYRAIKKK,26,8,-0.069231,1,0.076923,0.0,0.000000,0.0,...,0.000000,0.038462,0.038462,0.000000,0.153846,0.000000,0.038462,0.153846,0.038462,0.076923
3,12,RKRIHIGPGRAFYTT,15,5,-0.806667,1,0.066667,0.0,0.000000,0.0,...,0.000000,0.000000,0.066667,0.000000,0.200000,0.000000,0.133333,0.000000,0.000000,0.066667
4,14,GIWDTIKSMGKVFAGKILQNL,21,2,0.285714,1,0.047619,0.0,0.047619,0.0,...,0.047619,0.047619,0.000000,0.047619,0.000000,0.047619,0.047619,0.047619,0.047619,0.000000


### Izračunate značajke

Iz svake peptidne sekvence izračunate su numeričke značajke koje opisuju osnovna bioinformatička svojstva peptida. Duljina sekvence opisuje broj aminokiselina, neto naboj procjenjuje elektrostatska svojstva peptida, hidrofobnost opisuje sklonost interakciji s membranama, dok aminokiselinski sastav prikazuje udio svake od 20 standardnih aminokiselina u sekvenci.

In [ ]:
# Provjera dimenzija skupa značajki.
features_df.shape

(1672, 26)

In [ ]:
# Prikaz osnovne statistike glavnih značajki.
features_df[["length", "net_charge", "hydrophobicity"]].describe()

,length,net_charge,hydrophobicity
count,1672.000000,1672.000000,1672.000000
mean,21.490431,4.488636,-0.194518
std,12.771008,2.963551,1.066352
min,1.000000,-7.000000,-3.900000
25%,13.000000,3.000000,-0.774276
50%,19.000000,4.000000,-0.152985
75%,26.000000,6.000000,0.455332
max,119.000000,30.000000,3.250000


1672 peptida ✅
26 značajki ✅ (ID, SEQUENCE, length, net_charge, hydrophobicity, label + 20 AAC značajki)
Prosječna duljina ≈ 21,5 aminokiselina , što je očekivano za AMP.
Prosječan neto naboj ≈ +4.5 , što je u skladu s činjenicom da su AMP-ovi uglavnom kationski peptidi .
Prosječna hidrofobnost je blago negativna, ali s velikim rasponom, što je također očekivano.

In [ ]:
# Spremanje izračunatih značajki.
features_df.to_csv("/content/amp_features.csv", index=False)